app/scripts/lib/setupSentry.js

```diff
@@ -146,31 +154,88 @@ export default function setupSentry({ release, getState }) {
       } else {
         return null;
       }
-      return breadcrumb;
+      return rewriteBreadcrumb(breadcrumb);
     },
   });
 
-  function rewriteReport(report) {
-    try {
-      // simplify certain complex error messages (e.g. Ethjs)
-      simplifyErrorMessages(report);
-      // modify report urls
-      rewriteReportUrls(report);
-      // append app state
-      if (getState) {
-        const appState = getState();
-        if (!report.extra) {
-          report.extra = {};
-        }
-        report.extra.appState = appState;
+  return Sentry;
+}
+
+function hideUrlIfNotInternal(_url) {
+  const re = /^(chrome-extension|moz-extension):\/\//u;
+  if (!_url.match(re)) {
+    return '';
+  }
+  return _url;
+}
+
+export function rewriteBreadcrumb(breadcrumb) {
+  /** @todo check URL allowlist, exclude URLs not in allowlist */
+
+  if (breadcrumb.data?.url) {
+    breadcrumb.data.url = hideUrlIfNotInternal(breadcrumb.data.url);
+  }
+  if (breadcrumb.data?.to) {
+    breadcrumb.data.to = hideUrlIfNotInternal(breadcrumb.data.to);
+  }
+  if (breadcrumb.data?.from) {
+    breadcrumb.data.from = hideUrlIfNotInternal(breadcrumb.data.from);
+  }
+  return breadcrumb;
+}
+
+export function rewriteReport(report, getState) {
+  try {
+    // simplify certain complex error messages (e.g. Ethjs)
+    simplifyErrorMessages(report);
+    // remove urls from error message
+    sanitizeUrlsFromErrorMessages(report);
+    // Remove evm addresses from error message.
+    // Note that this is redundent with data scrubbing we do within our sentry dashboard,
+    // but putting the code here as well gives public visibility to how we are handling
+    // privacy with respect to sentry.
+    sanitizeAddressesFromErrorMessages(report);
+    // modify report urls
+    rewriteReportUrls(report);
+    // append app state
+    if (getState) {
+      const appState = getState();
+      if (!report.extra) {
+        report.extra = {};
       }
-    } catch (err) {
-      console.warn(err);
+      report.extra.appState = appState;
     }
-    return report;
+  } catch (err) {
+    console.error(err);
+    throw err;
   }
+  return report;
+}
 
-  return Sentry;
+function sanitizeUrlsFromErrorMessages(report) {
+  rewriteErrorMessages(report, (errorMessage) => {
+    let newErrorMessage = errorMessage;
+    const re =
+      /https?:\/\/(www\.)?[-a-zA-Z0-9@:%._+~#=]{1,256}\.[a-zA-Z0-9()]{1,63}\b([-a-zA-Z0-9()@:%_+.~#?&//=]*)/gu;
+    const urlsInMessage = newErrorMessage.match(re) || [];
+    urlsInMessage.forEach((url) => {
+      if (
+        !Object.values(ERROR_URL_ALLOWLIST).some((allowedUrl) =>
+          url.match(allowedUrl),
```

Comment:
It looks like `allowedUrl` is a string here, but `match` expects a regex. The docs for `match` suggest that when given a string, it implicitly creates a regex using the RegEx constructor. So this will give us a regex like this for codefi for example: `/codefi.network/`.

This isn't ideal in a number of ways:
* The unicode flag isn't used, so it might improperly handle unicode domains
* It will incorrectly allow URLs that have the allowed domains somewhere in the URL path
* This is generally a confusing way to use `match`, and we should instead explicitly create the regex when using match.

Reply:
changed the approach here, to match against the URL hostname https://github.com/MetaMask/metamask-extension/pull/16780/commits/daaac950566477c5197089c509babada82f9cdaa#diff-a8c55d1ad0724e946c8c35114d6a298b74904accdddedff77e1b8ade2097fd65R251-R256